lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -599,9 +597,63 @@ sub get_numeric_from_hex {
 
     # numeric responses should have at least 64 + 2(0x) = 66 characters
     # transaction data field / contract response
-    return undef unless ($hex && length($hex) == 66);
+    return undef
+        unless ($hex && $hex =~ /^0x[0-9a-fA-F]{64}$/);
 
     return Math::BigFloat->from_hex($hex);
 }
 
+=head2 get_valid_decimal_places
+
+Check the given decimal places contract response following the steps:
+- hex is valid
+- hex is a numeric value
+- is a valid Math::BigFloat not 0
+- is between the limit of 18 decimal places
+
+=over 4
+
+=item * C<decimal_places> contract decimal places
```

Comment:
This argument name, and the name of the sub itself, are very confusing. Shouldn't the arg be "hex" and the sub something like "hex_is_valid_number"?

Reply:
I already have the `get_numeric_from_hex` here in this class, is being used inside of these functions, also `is_something` will give the impression this function is a bool return, I'm using `get` here because it is going to return the valid object or undef, but I agree that the arguments should have the word hex in them I will add it